src/gcovr/__main__.py

```diff
@@ -209,7 +209,7 @@ def load_config(partial_options) -> Dict[str, Any]:
     return {}
 
 
-def main(args=None):
+def main(args: Optional[List[str]] = None) -> int:
     """The main entry point of GCOVR."""
```

Comment:
I wanted to do this with a separate PR. I saw also that the entry point is wrapped in a `sys.exit` and always using a return from main would simplify the usage of gcovr main inside other scripts.

Reply:
No need for a separate class. Calling `sys.exit(x)` is pretty much just an alias for `raise SystemExit(x)`, which can be caught as normal. Roughly:

```python
try:
    argparse_stuff()
except SystemExit sys_exit:
    if sys_exit.args:
        return sys_exit.args[0]
    return 0
```

Reply:
Ok, but we need to remap the exit code because argparse uses 2 for an argument error and 0 if help output is requested. Shall we rely on this hardcoded 2 in the error method or is it not better to use a separate class?

Reply:
Really good that you found this conflict between the usage of exit status 2 for `--fail-under-line` vs argparse!

The "official" way to handle this (in Python 3.9 or later) is to use [exit_on_error=False](https://docs.python.org/3/library/argparse.html#exit-on-error). I'd be inclined to just use that feature, instead of implementing a workaround for EOL Pythons.

Reply:
Then I'll change this with the drop of support for 3.8.